lib/google/gax/operation.rb

```diff
@@ -161,6 +172,29 @@ def done?
         @grpc_op.done
       end
 
+      # Checks if the operation is done and the result is an response.
+      # If the operation is not finished then this will return false.
+      #
+      # @return [Boolean] Whether an error has been returned.
+      def response?
+        done? ? @grpc_op.result == :response : false
+      end
+
+      # If the operation is done, returns the response, otherwise returns nil.
+      # If a type is provided, the response will be unpacked using the type
+      # provided; returning nil if the response is not of the type provided.
+      # If the type is not of provided, the response will be unpacked using
+      # the response's type_url if the type_url is found in the
+      # Google::Protobuf::DescriptorPool.generated_pool.
+      # If the type cannot be found the raw response is retuned.
+      #
+      # @return [Object, nil]
+      #   The response of the operation.
+      def response
+        return unpack(@grpc_op.response, @result_type) if response?
+        nil
```

Comment:
This is me being overly cautious and verbose, because not everyone knows the behavior of the guard statement. This is equivalent to the following:

```ruby
def response
  unpack(@grpc_op.response, @result_type) if response?
end
```

I've trim the fat in these methods .